### ISSUE TITLE
race condition on concurrent cache misses

### ISSUE THREAD

**Issue by bukzor** [Yelp/venv-update#155]
```
15:02:39.075 Downloading/unpacking closure-tools==6.8.0 (from -r requirements.txt (line 50))
15:02:39.999 The file ~/.cache/pip-faster/wheelhouse/closure_tools-6.8.0-py2.py3-none-any.whl exists. (i)gnore, (w)ipe, (b)ackup Cleaning up...
15:02:40.089 Exception:
15:02:40.090 Traceback (most recent call last):
15:02:40.090   File "venv/local/lib/python2.7/site-packages/pip/basecommand.py", line 122, in main
15:02:40.090     status = self.run(options, args)
15:02:40.090   File "venv/local/lib/python2.7/site-packages/pip_faster.py", line 475, in run
15:02:40.090     requirement_set = super(FasterInstallCommand, self).run(options, args)
15:02:40.090   File "venv/local/lib/python2.7/site-packages/pip/commands/install.py", line 278, in run
15:02:40.090     requirement_set.prepare_files(finder, force_root_egg_info=self.bundle, bundle=self.bundle)
15:02:40.090   File "venv/local/lib/python2.7/site-packages/pip_faster.py", line 387, in prepare_files
15:02:40.091     super(FasterRequirementSet, self).prepare_files(finder, **kwargs)
15:02:40.091   File "venv/local/lib/python2.7/site-packages/pip/req.py", line 1197, in prepare_files
15:02:40.091     do_download,
15:02:40.091   File "venv/local/lib/python2.7/site-packages/pip/req.py", line 1375, in unpack_url
15:02:40.091     self.session,
15:02:40.091   File "venv/local/lib/python2.7/site-packages/pip/download.py", line 578, in unpack_http_url
15:02:40.091     _copy_file(temp_location, download_dir, content_type, link)
15:02:40.092   File "venv/local/lib/python2.7/site-packages/pip/download.py", line 456, in _copy_file
15:02:40.092     display_path(download_location), ('i', 'w', 'b'))
15:02:40.092   File "venv/local/lib/python2.7/site-packages/pip/util.py", line 124, in ask_path_exists
15:02:40.092     return ask(message, options)
15:02:40.092   File "venv/local/lib/python2.7/site-packages/pip/util.py", line 132, in ask
15:02:40.092     response = raw_input(message)
15:02:40.092 EOFError: EOF when reading a line
15:02:40.092 
15:02:40.093 Storing debug log for failure in ~/.pip/pip.log
15:02:40.112 
15:02:40.114 Something went wrong! Sending 'venv' back in time, so make knows it's invalid.
15:02:40.114 Running virtualenv with interpreter /usr/bin/python2.7
15:02:40.119 Command exited with non-zero status 2
15:02:40.119 2.99user 0.69system 0:04.59elapsed 80%CPU (0avgtext+0avgdata 137584maxresident)k
```

cc @chriskuehl  @ydgholz

**Comment by asottile** [comment]
It's easier to force this race with larger packages

**Comment by chriskuehl** [comment]
This seems to reliably repoduce it:

```bash
rm -rf venv1 venv2 venv_update.py ~/.cache/pip-faster
wget -O venv_update.py https://raw.githubusercontent.com/Yelp/venv-update/v1.1.0/venv_update.py

python3.5 venv_update.py venv= venv1 install= django==1.6 &
python3.5 venv_update.py venv= venv2 install= django==1.6 &
wait
```

**Comment by asottile** [comment]
This is fixed if you change the script to 2.0.0!!!

```bash
#!/usr/bin/env bash
export XDG_CACHE_HOME="$PWD/CACHE"
rm -rf venv1 venv2 venv_update.py "$XDG_CACHE_HOME/pip-faster"
wget -O venv_update.py https://raw.githubusercontent.com/Yelp/venv-update/v2.0.0/venv_update.py

python3.5 venv_update.py venv= venv1 install= django==1.6 &
python3.5 venv_update.py venv= venv2 install= django==1.6 &
python3.5 venv_update.py venv= venv3 install= django==1.6 &
python3.5 venv_update.py venv= venv4 install= django==1.6 &
python3.5 venv_update.py venv= venv5 install= django==1.6 &
python3.5 venv_update.py venv= venv6 install= django==1.6 &
wait
```
